pkg/tracking/store/sql/tags.go

```diff
@@ -80,3 +81,182 @@ func (s TrackingSQLStore) setTagsWithTransaction(
 
 	return nil
 }
+
+const (
+	maxEntityKeyLength = 250
+	maxTagValueLength  = 8000
+)
+
+// Helper function to validate the tag key and value
+func validateTag(key, value string) *contract.Error {
+	if key == "" {
+		return contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			"Missing value for required parameter 'key'",
+		)
+	}
+	if len(key) > maxEntityKeyLength {
+		return contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			fmt.Sprintf("Tag key '%s' had length %d, which exceeded length limit of %d", key, len(key), maxEntityKeyLength),
+		)
+	}
+	if len(value) > maxTagValueLength {
+		return contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			fmt.Sprintf("Tag value exceeded length limit of %d characters", maxTagValueLength),
+		)
+	}
+	// TODO: Check if this is the correct way to prevent invalid values
+	if _, err := strconv.ParseFloat(value, 64); err == nil {
+		return contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			fmt.Sprintf("Invalid value %s for parameter 'value' supplied", value),
+		)
+	}
+	return nil
+}
+
+func (s TrackingSQLStore) SetTag(
+	ctx context.Context, runID, key, value string,
+) *contract.Error {
+	if runID == "" {
+		return contract.NewError(
+			protos.ErrorCode_INVALID_PARAMETER_VALUE,
+			"RunID cannot be empty",
+		)
+	}
+
+	// If the runID can be parsed as a number, it should throw an error
+	if _, err := strconv.ParseFloat(runID, 64); err == nil {
```

Comment:
This again is weird, where is this coming from? Feel very unnecessary.

Reply:
This is also due to some of the weird edge cases that I thought were failing. This is what the previous Python tests looked like:

```python
def assert_bad_request(payload, expected_error_message):
      response = _send_rest_tracking_post_request(
          mlflow_client.tracking_uri,
          "/api/2.0/mlflow/runs/set-tag",
          payload,
      )
      assert response.status_code == 400
      assert expected_error_message in response.text

assert_bad_request(
  {
      "run_id": 31,
      "key": "tag",
      "value": 41,
  },
  "Invalid value 31 for parameter 'run_id' supplied",
)
assert_bad_request(
  {
      "run_id": run_id,
      "key": "param",
      "value": 41,
  },
  "Invalid value 41 for parameter 'value' supplied",
)
assert_bad_request(
  {
      "run_id": run_id,
      # Missing key
      "value": "value",
  },
  "Missing value for required parameter 'key'",
)

response = _send_rest_tracking_post_request(
  mlflow_client.tracking_uri,
  "/api/2.0/mlflow/runs/set-tag",
  {
      "run_uuid": run_id,
      "key": "key",
      "value": "value",
  },
)
```

Reply:
It seems that IDs are expected to be in the UUID form rather than simple integer form